### ISSUE TITLE
gapminder_dashboard - Group 5

### ISSUE THREAD

**Issue by mel-liow** [UBC-MDS/DSCI532-peer-review#11]
# DSCI532-peer-review

### To complete by the group
* Name of the app:  Gapminder Dashboard
* Link to the repo in milestone2:  https://github.com/UBC-MDS/gapminder_dashboard
* Link to the `proposal.md` file where it is the description of the persona and scenario:  https://github.com/UBC-MDS/gapminder_dashboard/blob/main/docs/proposal.md

### To complete by the group reviewing:

**Persona**
* Provide 3 pieces of constructive feedback that could help the app to be better suited to the scenario it was designed for
* Provide 3 pieces of constructive feedback that could help the app to be better suited to the user(persona) it was designed for

**Frontend / layout**
* Is the text (plot labels / titles / subtitles) easy to read? 
* Are the plots selected appropiated for the data? Can you think of other alternatives or improvements for the plots regarding interactivity/reactivity or plot selection?
* Was the dashboard organized in a way that was easy to read (columns / rows) or you would suggest a change in this sense?
* Would you prefer including tabs / navbars to improve the navigation or you think it is better to visualize everything in one screen by the nature of the content?
* Are the plots or other images described properly for the persona described?

**Backend**
* Are the options that you can select in the widgets working fine or have you found some cases where it is not working?
* If there are interactive plots: are the labels clear and easy to understand?

**Documentation**
* Was the documentation easy to understand from the `README.md` how to use the app? How could the documentation be improved?
* Was the app intuitive to use without reading the documentation? What information could be good to include? 

**Indicate something you would like to include in your own app after reviewing this one**

---

**Comment by morganrosenberg50** [comment]
_Note: you identify a few use cases here, one being around teaching statistics, and the other around communicating insights and educating about historical social/economics trends. Given that your user is a high school economics teacher, I’ll focus this review on the latter._ 

Also, great work overall! It’s very fun to play around with and informative. Everything included below are thoughts on how to make it even better. In reviewing this app, I was inspired by the bubble chart, which could be a very powerful way for us to represent our own data which some of which reports on different types of employees’ scores at a national and regional level. Thanks for the inspiration! 

### Persona - Scenario 

1. Given that this dashboard is going to be used to understand trends, it would be helpful if the axis scale matched the data – for example many versions of the chart generated in the bottom right “Target 1 vs Target 2” for population density vs population are all within the 0-2,000 range where as the y axis ranges from 0-9000. While this is the most extreme example, there are other analogous ones.
2. Especially for teachers looking to identify specific trends or differences over time, it may be helpful to be able to directly select two countries for comparison (e.g. in the target of study over time). 
3. Especially to address misconceptions, it may be helpful to be able to also have the option to view these trends over time (e.g. a slider for year to quickly toggle across a time horizon). 

### Persona – User

1. It may also be useful to include some interesting settings for your drop-down menus – right now showing interesting insights requires a degree of trial and error. Maybe you could include in your readme you include a few common lessons and the best settings to illustrate them.
2. Although you write the filter variables in plain English, for students to be able to more fully appreciate the charts, it may be helpful to share their precise definitions at the bottom of the dashboard similar to how you define them in the readme.
3. Given that your user teaches high school students, and is likely looking to make the subject matter “come alive” for students, it may be useful to make the aesthetic more ‘fun’ with different colours or fun titles (e.g. for your top 10 countries chart having custom titles like “Who’s the richest?”, “Who’s the biggest?”, for each of your filters). 

### Frontend/Layout

- Text is overall easy to read and clear. For your target of study over time chart, when lines converge, their labels overlap, however, I think it is usually still fairly clear which line is which given that the user selects the country and region of interest
- Consider adapting the bubble plot as described in scenario.
- The tooltip application on your first chart is great!
- Otherwise, charts seem appropriate and effective
- It took me some time to understand which filters applied to which charts. Although it is intuitive once you pause and understand what you’re looking at, it would improve the ux to indicate this explicitly somewhere for filters that are meant for only one chart
- I think it’s powerful to have all four charts on one screen as you have it organized presently (no need for tabs)
- In addition to the suggestion of “fun titles” where doable, the titles for Selected Region, Target of study over time, and Target 1 vs Target two could be more informative 

### Backend 
- Your first chart breaks when region = all; in the short term, consider changing the default value to Africa until it is fixed
- Otherwise everything worked as expected! 

### Documentation
- Documentation was detailed and effective. To take it to the next level, consider adding distinctly variable settings to meet specific use cases (e.g. to illustrate the link between gdp and health, look at gdp and child mortality within X date range and Y regions)
- The app was very intuitive to use, for anyone familiar with this type of data, and only required the documentation for the purposes of this review.
